src/wrapper.rs

```diff
@@ -50,23 +50,27 @@ impl DxcBlob {
         Self { inner }
     }
 
-    pub fn from_slice(slice: &[u8]) -> Self {
+    pub fn from_slice(slice: &[u8]) -> Result<Self, HassleError> {
         let mut blob = ComPtr::<IDxcBlob>::new();
 
         // This weird pointer cast from IDxcBlob to ID3DBlob is safe:
         // See https://github.com/microsoft/DirectXShaderCompiler/blob/285490211f2cbb44444a23d5f1c329881b96a43b/docs/HLSLChanges.rst
         // `The HLSL compiler avoids pulling in DirectX headers and defines an
         // IDxcBlob interface that has the same layout and interface identifier (IID).`
         let blob_ptr: *mut *mut IDxcBlob = blob.as_mut_ptr::<IDxcBlob>();
-        unsafe {
-            winapi::um::d3dcompiler::D3DCreateBlob(slice.len(), blob_ptr as *mut *mut _);
-            std::ptr::copy_nonoverlapping(
-                slice.as_ptr(),
-                blob.get_buffer_pointer() as *mut u8,
-                slice.len(),
-            );
-        }
-        Self { inner: blob }
+        return_hr_wrapped!(
+            unsafe {
+                let hr =
+                    winapi::um::d3dcompiler::D3DCreateBlob(slice.len(), blob_ptr as *mut *mut _);
+                std::ptr::copy_nonoverlapping(
+                    slice.as_ptr(),
+                    blob.get_buffer_pointer() as *mut u8,
+                    slice.len(),
+                );
+                hr
+            },
+            Self { inner: blob }
+        );
```

Comment:
It's dangerous copy something into `blob` when `D3DCreateBlob` returns an error (consequently, `get_buffer_pointer()` likely returns 0). Instead, it should happen in the sucess-expresssion of `return_hr_wrapped`:

```suggestion
        return_hr_wrapped!(
            unsafe {
                return_hr_wrapped!(
                winapi::um::d3dcompiler::D3DCreateBlob(slice.len(), blob_ptr as *mut *mut _)
            },
            unsafe {
                std::ptr::copy_nonoverlapping(
                    slice.as_ptr(),
                    blob.get_buffer_pointer() as *mut u8,
                    slice.len(),
                );

                Self { inner: blob }
            }
        );
```

Probably put `unsafe` around the entire macro invocation though.